### PR TITLE
[Merged by Bors] - feat(MeasureTheory): remove an `AbsolutelyContinuous` hypothesis from `inv_rnDeriv`

### PR DESCRIPTION
**Pull request by RemyDegenne**
In order to remove that hypothesis, we also:
- add some basic lemmas about absolute continuity and mutually singular measures.
- add HaveLebesgueDecomposition instances
- rewrite the proof of `withDensity_rnDeriv_eq` to use the new API instead of unfolding the definitions
- generalize `rnDeriv_restrict` and `rnDeriv_withDensity` to possibly different measures

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Decomposition/RadonNikodym.lean [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /-
 Copyright (c) 2021 Kexing Ying. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Kexing Ying
+Authors: Kexing Ying, Rémy Degenne
 -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Decomposition/RadonNikodym.lean:5]: Could you mention your additions to the file-level docstring?

File: Mathlib/MeasureTheory/Measure/MutuallySingular.lean [hunk 2]
```diff
@@ -53,6 +53,23 @@ theorem mk {s t : Set α} (hs : μ s = 0) (ht : ν t = 0) (hst : univ ⊆ s ∪
...
+lemma measure_nullSet (h : μ ⟂ₘ ν) : μ h.nullSet = 0 := h.choose_spec.2.1
+
+@[simp]
+lemma measure_compl_nullSet (h : μ ⟂ₘ ν) : ν h.nullSetᶜ = 0 := h.choose_spec.2.2
+
+@[simp]
+lemma restrict_nullSet (h : μ ⟂ₘ ν) : μ.restrict h.nullSet = 0 := by simp
+
+@[simp]
+lemma restrict_compl_nullSet (h : μ ⟂ₘ ν) : ν.restrict h.nullSetᶜ = 0 := by simp
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MutuallySingular.lean:71]: I don't think those should be simp lemmas, as they can be proved directly by simp.

**Review by sgouezel — COMMENTED** [review]
This is great, thanks a lot!
bors d+

File: Mathlib/MeasureTheory/Measure/MutuallySingular.lean (hunk 2 above)
Comment (RemyDegenne) [reply Mathlib/MeasureTheory/Measure/MutuallySingular.lean:71]: It indeed looks like they should not be simp lemmas for that reason, but simp does not use them if I don't tag them, and the linter does not complain.
I'll add a note about that strange situation.

**Pull request closed without merging**